restore CSE'd node metadata in runtime asserts pass

pianpwk opened the pull request:
Adds val, and optionally stack_trace & nn_module_stack metadata back to SymInt compute nodes that we CSE, with a hook on `graph.create_node()`. Not sure if there's other metadata we want to populate here?




cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

pianpwk committed:
[export] restore CSE'd node metadata

Differential Revision: D61822108

pianpwk committed:
Merge branch 'main' of https://github.com/pytorch/pytorch into export-D61822108

pianpwk committed:
Merge branch 'main' of https://github.com/pytorch/pytorch into export-D61822108

pianpwk committed:
lint

pianpwk commented on torch/fx/passes/runtime_assert.py:
```diff
@@ -145,6 +147,24 @@ def _is_intermediate_tensor_sym_call(node: fx.Node) -> bool:
         )
 
+    # Figure out if we're in Dynamo or AOTAutograd
+    val_key = "val" if export else "example_value"
```
Not sure what the best way is to figure out which key to use. Maybe we could iterate through nodes and check what's present? And then fallback to this flag if that's empty for some reason.

ezyang replied:
I usually just make a helper that fetches either val or example_value, depending on which one is set, preferring val. There's one in symbolic_shapes

pianpwk committed:
restore CSE'd node metadata in runtime asserts pass (#134516)

Summary:
Adds val, and optionally stack_trace & nn_module_stack metadata back to SymInt compute nodes that we CSE, with a hook on `graph.create_node()`. Not sure if there's other metadata we want to populate here?




cc XilunWu H-Huang awgu kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o


Reviewed By: ezyang

Differential Revision: D61822108

Pulled By: pianpwk

pianpwk commented:
@pytorchbot revert -m "breaking NestedTensor test" -c ghfirst

pianpwk committed:
mergre

pianpwk committed:
Merge branch 'main' of https://github.com/pytorch/pytorch into export-D61822108

pianpwk committed:
catch NotImplemented

pianpwk committed:
bad merge

pianpwk commented:
@pytorchbot merge

ezyang commented:
This PR regressed instruction count on update_hint_regression from 10.59923 B to 10.72784 B.

<img width="366" alt="image" src="https://github.com/user-attachments/assets/7f76c669-1960-4f17-8bb7-fd2669f5a427">

The recovery in this chart was caused by an unrelated optimization https://github.com/pytorch/pytorch/pull/135256
